src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -82,13 +82,35 @@ public static function fromDsn(string $dsn, array $options = array(), bool $debu
         }
 
         $exchangeOptions = $amqpOptions['exchange'];
-        $queueOptions = $amqpOptions['queue'];
+        $queueOptions = self::getQueueOptions($amqpOptions['queue']);
 
         unset($amqpOptions['queue'], $amqpOptions['exchange']);
 
         return new self($amqpOptions, $exchangeOptions, $queueOptions, $debug, $amqpFactory);
     }
 
+    private static function getQueueOptions(array $queueOptions): array
+    {
+        if (!array_key_exists('arguments', $queueOptions)) {
```

Comment:
[tip] You can you `isset()` instead of  `array_key_exists` here, you will have the same behaviour without the cost for a function call.

Reply:
So, can we keep `array_key_exists` @alquerci ?

Reply:
:+1: